juju/juju
# feat: state get implementations

Adds state get implementations, the set charm is going to be in subsequent PR, as I believe that it will take a lot of time to correctly implement. Manifest, actions, config, and LXD profiles are row-based structs, which is the normal form, unlike charm metadata. The manifest bases are defined as an array and an index was added to the state schema to ensure correct ordering was maintained (although we do run into another sqlair bug[1]). Architectures will not be guaranteed to be in order, because we're storing the base array and the architectures as one flat row. If we did want to do that, we should create another table that kept track of that one as well. For now, the order of architectures shouldn't matter as we don't pick one over the other.

 1. https://github.com/canonical/sqlair/issues/155





## Checklist




## QA steps

The state tests should pass. Once the `SetCharm` is implemented in subsequent PRs then we should start to see full end to end tests, which would better validate all the state implementation.


## Links


**Jira card:** [JUJU-6015](https://warthogs.atlassian.net/browse/JUJU-6015)



[JUJU-6015]: https://warthogs.atlassian.net/browse/JUJU-6015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
